source/macos/AdaptiveCards/AdaptiveCards/Managers/ACSVisibilityManager.swift

```diff
@@ -0,0 +1,167 @@
+//
+//  ACSVisibilityManager.swift
+//  AdaptiveCards
+//
+//  Created by uchauhan on 30/08/22.
+//
+//  This class is designed to handle the visibility of views as well as the corresponding padding or separator. Class also depended on Filler Space Manager to find associated views.
+
+import AdaptiveCards_bridge
+import AppKit
+
+/// VisibilityManager protocols for handle element visibility
+@objc protocol ACSIVisibilityManagerFacade {
+    func hideView(_ view: NSView)
+    func unhideView(_ view: NSView)
+}
+
+class ACSVisibilityManager {
+    /// tracks objects that are used in filling the space
+    private var fillerSpaceManager: ACSFillerSpaceManager
+    
+    /// tracks visible views
+    private var visibleViews: NSMutableOrderedSet
+    
+    var hasVisibleViews: Bool { return !visibleViews.set.isEmpty }
+    
+    init(_ fillerSpaceManager: ACSFillerSpaceManager) {
+        self.fillerSpaceManager = fillerSpaceManager
+        self.visibleViews = NSMutableOrderedSet()
+    }
+    
+    /// adds index of a visible view to a visible views collection, and the index is maintained in sorted order
+    /// in increasing order.
+    func addVisibleView(_ index: Int) {
+        let indexAsNumber = NSNumber(value: index)
+        if visibleViews.contains(indexAsNumber) { return }
+        let range = NSRange(location: 0, length: visibleViews.count)
+        let insertionIndex = visibleViews.index(of: indexAsNumber, inSortedRange: range, options: .insertionIndex, usingComparator: { num0, num1 in
+            if let n1 = num0 as? NSNumber, let n2 = num1 as? NSNumber {
+                return n1.compare(n2)
+            }
+            return .orderedSame
+        })
+        visibleViews.insert(indexAsNumber, at: insertionIndex)
+    }
+    
+    /// removes the index of view from the visible views collection
+    /// maintain the sorted order after the removal
+    func removeVisibleView(at index: Int) {
+        let indexAsNumber = NSNumber(value: index)
+        if !visibleViews.contains(indexAsNumber) { return }
+        let range = NSRange(location: 0, length: visibleViews.count)
+        let removalIndex = visibleViews.index(of: indexAsNumber, inSortedRange: range, options: .insertionIndex, usingComparator: { num0, num1 in
+            if let n1 = num0 as? NSNumber, let n2 = num1 as? NSNumber {
+                return n1.compare(n2)
+            }
+            return .orderedSame
+        })
+        visibleViews.removeObject(at: removalIndex)
+    }
+    
+    /// YES means the index of view is currently the leading or top view
+    func amIHead(_ index: Int) -> Bool {
```

Comment:
Prefer Boolean to start with *is*, like isFirstElement

Reply:
Ok ✅